### PR TITLE
1756 pid monitoring

### PR DESCRIPTION
**Pull request by jessebs**
Pull request to add status reporting while using pids.

I had some tests that I ended up reverting.  TestPumaControlCli#test_control_url interfered with any other threaded tests and the fact that signals are used to kill the puma process means the tests were self destructing.

Testing was doable using forked processes but they aren't supported by windows.

### CODE REVIEW

File: lib/puma/control_cli.rb [hunk 1]
```diff
@@ -121,7 +121,7 @@ def prepare_configuration
         @control_url = sf.control_url
         @control_auth_token = sf.control_auth_token
         @pid = sf.pid
-      elsif @pidfile
+      elsif @pidfile && File.readable?(@pidfile)
```
Comment (nateberkopec) [inline lib/puma/control_cli.rb]: Why was this added?

**Comment by nateberkopec** [comment]
>TestPumaControlCli#test_control_url interfered with any other threaded tests and the fact that signals are used to kill the puma process means the tests were self destructing.

Hm :(

**Comment by nateberkopec** [comment]
@jessebs 👋

File: lib/puma/control_cli.rb (hunk 1 above)
Comment (spk) [reply lib/puma/control_cli.rb]: o/ this was added to avoid `No such file or directory @ rb_sysopen - tmp/pids/server.pid` when defined in `config/puma.rb` and have instead `Neither pid nor control url available` from https://github.com/puma/puma/blob/v3.12.1/lib/puma/control_cli.rb#L183

Comment (nateberkopec) [reply lib/puma/control_cli.rb]: I think this could be a separate PR-plus-tests, then.

Comment (spk) [reply lib/puma/control_cli.rb]: > I think this could be a separate PR-plus-tests, then.

First PR done in https://github.com/puma/puma/pull/1824

**Comment by nateberkopec** [comment]
Closing for #1824
